### PR TITLE
docs: add complete example for canStartBuild

### LINKED ISSUE #3899: Prevent use of a Worker is Load Average is too high
nextWorker doesn't seem to have the ability. An example is that 4 buildrequests will always be distributed to 4 workers, one for each. While the default implementation will randomly distribute them which can not asure that.

### PR DESCRIPTION
**Pull request by rjarry**
Fixes #3899

### CODE REVIEW

**Comment by rjarry** [comment]
@s0undt3ch let me know if that is enough :)

File: master/docs/manual/customization.rst [hunk 1]
```diff
@@ -256,7 +256,72 @@ If the file does not exist on the given worker, refuse to run the build to force
...
+   @defer.inlineCallbacks
+   def shell(command, worker, builder):
+       args = {
+           'command': command,
+           'logEnviron': False,
+           'workdir': worker.worker_basedir,
+           'want_stdout': False,
+           'want_stderr': False,
+       }
+       cmd = RemoteCommand(name, args, stdioLogName=None)
```
Comment (s0undt3ch) [inline master/docs/manual/customization.rst]: `RemoteCommand(command, args, stdioLogName=None)` ? Or, where does `name` come from?

Comment (rjarry) [reply master/docs/manual/customization.rst]: oops my bad

**Comment by s0undt3ch** [comment]
I hit a traceback when receiving the remote command payload. Can't debug it today though. Will try tomorrow morning.

Something a out None not having a remote something :) not helpful I know.

**Comment by s0undt3ch** [comment]
```
2018-06-27 03:39:51-0600 [Broker,36,10.1.51.2] Peer will receive following PB traceback:
2018-06-27 03:39:51-0600 [Broker,36,10.1.51.2] Unhandled Error
        Traceback (most recent call last):
          File "/home/buildbot/.pyenv/versions/2.7.13/envs/Buildbot/lib/python2.7/site-packages/twisted/spread/banana.py", line 173, in gotItem
            self.callExpressionReceived(item)
          File "/home/buildbot/.pyenv/versions/2.7.13/envs/Buildbot/lib/python2.7/site-packages/twisted/spread/banana.py", line 136, in callExpressionReceived
            self.expressionReceived(obj)
          File "/home/buildbot/.pyenv/versions/2.7.13/envs/Buildbot/lib/python2.7/site-packages/twisted/spread/pb.py", line 621, in expressionReceived
            method(*sexp[1:])
          File "/home/buildbot/.pyenv/versions/2.7.13/envs/Buildbot/lib/python2.7/site-packages/twisted/spread/pb.py", line 1025, in proto_message
            self._recvMessage(self.localObjectForID, requestID, objectID, message, answerRequired, netArgs, netKw)
        --- <exception caught here> ---
          File "/home/buildbot/.pyenv/versions/2.7.13/envs/Buildbot/lib/python2.7/site-packages/twisted/spread/pb.py", line 1054, in _recvMessage
            netResult = object.remoteMessageReceived(self, message, netArgs, netKw)
          File "/home/buildbot/.pyenv/versions/2.7.13/envs/Buildbot/lib/python2.7/site-packages/twisted/spread/flavors.py", line 120, in remoteMessageReceived
            state = method(*args, **kw)
          File "/home/buildbot/.pyenv/versions/2.7.13/envs/Buildbot/lib/python2.7/site-packages/buildbot/process/remotecommand.py", line 196, in remote_update
            self.worker.messageReceivedFromWorker()
        exceptions.AttributeError: 'NoneType' object has no attribute 'messageReceivedFromWorker'
```

**Comment by rjarry** [comment]
@s0undt3ch I forgot to set cmd.worker !

Thanks for testing :)

**Pull request merged**